share/doc/homebrew/Maintainer-Guidelines.md

```diff
@@ -6,20 +6,118 @@ definitely not a beginner’s guide.
 
 Maybe you were looking for the [Formula Cookbook](Formula-Cookbook.md)?
 
+----
+
+Quick links:
+  [General Guidelines](#general-guidelines) •
+  [Core and Core PRs](#core-and-core-prs) •
+  [Formulae and Formula PRs](#formulae-and-formula-prs)
+
+## General Guidelines
+
 ## Quick Checklist
+1.  [Ensure you have set your username and email address
+    properly](https://help.github.com/articles/setting-your-email-in-git/).
+2.  Sign off cherry-picks if you amended them
+    ([GitX-dev](https://github.com/rowanj/gitx) can do this, otherwise there is
+    a command line flag for it).
+3.  If the commit fixes a bug, use “Fixes \#104” syntax to close the bug
+    report and link to the commit.
+
+### Add Comments
+It may be enough to refer to an issue ticket, but make sure changes that would
+make sense to you, if you came to them unaware of the surrounding issues. Many
+times on other projects I’ve seen code removed because the new guy didn’t know
```

Comment:
Yeah, this has been just rearranged, but not edited further. I'll make sure to review and update the entire text before my next push.